webapp/niapp.yaml

```diff
@@ -0,0 +1,416 @@
+      description: Get webapp metadata by Id
+      operationId: get-webapp
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          $ref: '#/responses/GetWebAppResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    put:
+      tags:
+        - metadata
+      summary: Update existing webapp metadata
+      description: Update webapp metadata by Id
+      operationId: update-webapp
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/UpdateWebAppRequest'
+      responses:
+        200:
+          $ref: '#/responses/UpdateWebAppResponse'
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    delete:
+      tags:
+        - metadata
+      summary: ''
+      description: Delete webapp by id
+      operationId: delete-webapp
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/webapps':
+    get:
+      tags:
+        - metadata
+      summary: Lists the webapps
+      description: Lists the webapps
+      operationId: list-webapps
+      parameters:
+        - in: query
+          name: name
+          description: Filters the webapps by name
+          type: string
+        - in: query
+          name: type
+          description: Filters the webapps by type
+          type: string
+          enum:
+          - Dashboard
+          - DashboardTemplate
+          - WebVI
+        - in: query
+          name: userId
+          description: Filters the webapps by the user id of the owner
+          type: string
+        - in: query
+          name: workspace
+          description: Filters the webapps by workspace id
+          type: string
+        - in: query
+          name: continuationToken
+          description: The continuation token can be used to paginate through the webapp list results. Provide a token to continue a previous listing.
+          type: integer
+        - in: query
+          name: take
+          description: How many webapps to return in the result
+          type: integer
+          default: 100
+          maximum: 1000
+      responses:
+        200:
+          $ref: '#/responses/ListWebAppsResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+    post:
+      tags:
+        - metadata
+      summary: Creates a new webapp
+      description: Creates a new webapp
+      operationId: create-webapp
+      parameters:
+        - $ref: '#/parameters/CreateWebAppRequest'
+      responses:
+        200:
+          $ref: '#/responses/CreateWebAppResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+  '/webapps/{id}/sharing':
+    put:
```

Comment:
Suggestion, feel free to ignore: For API completeness, we might also want to consider adding a GET /webapps/{id}/sharing in addition to the update route. Especially since the other properties below (shared-emails, content, etc.) have a corresponding GET route as well.

Reply:
This is the documentation of the existing implementation.